### PR TITLE
Bugfix/456 ic select searchable default value

### PR DESCRIPTION
**Pull request by jd239**
## Summary of the changes
- Default value appears for IcSelect searchable
- When initial value is set and there are no results, the menu does not display
- Check for searchResultsEl before applying innerText
- Set value and option value to lowercase so they match when typing value. This was not working when matching all uppercase value with lowercase.

## Related issue
#456 

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [ ] I have manually accessibility tested any changes, if relevant.
- [ ] I have ensured any changes match the Figma component library.

### CODE REVIEW

**Review by GCHQ-Developer-465 — APPROVED** [review]

File: packages/web-components/src/components/ic-select/ic-select.spec.tsx [hunk 1]
```diff
@@ -1199,4 +1200,38 @@ describe("ic-select", () => {
...
+    page.root.options = [];
+    await page.waitForChanges();
+
+    expect(page.rootInstance.searchableSelectInputValue).toBe("Test value 01");
+
+    const input = page.root.shadowRoot.querySelector("input");
+    expect(input.value).toBe("Test value 01");
+  });
+
+  it("should set the default value of searcahable as option label if matching label/value exists", async () => {
```
Comment (ad9242) [inline packages/web-components/src/components/ic-select/ic-select.spec.tsx:1219]:
```suggestion
  it("should set the default value of searchable as option label if matching label/value exists", async () => {
```

File: packages/web-components/src/components/ic-select/ic-select.e2e.ts [hunk 2]
```diff
@@ -1471,6 +1481,15 @@ describe("ic-select", () => {
     expect(value).toBe(undefined);
   });
 
+  it("shouldnt display no results found on initial load and setting default value on disable filter", async () => {
```
Comment (ad9242) [inline packages/web-components/src/components/ic-select/ic-select.e2e.ts:1484]:
```suggestion
  it("shouldn't display no results found on initial load if setting default value and disable filter", async () => {
```

**Review by ad9242 — COMMENTED** [review]
a couple of little suggestions

File: packages/web-components/src/components/ic-select/ic-select.e2e.ts (hunk 2 above)
Comment (GCHQ-Developer-465) [reply packages/web-components/src/components/ic-select/ic-select.e2e.ts:1484]: Perhaps it should be 'should not' for consistency as most e2e tests are using 'should not'.

**Review by GCHQ-Developer-465 — CHANGES_REQUESTED** [review]

File: packages/react/src/stories/ic-select.stories.mdx [hunk 3]
```diff
@@ -489,6 +507,32 @@ export const searchableOptionsWithRecommended = [
   </Story>
 </Canvas>
 
+export const SearchableDefaultValue = () => {
```
[the remark this answers is not in the document]
Comment (jd239) [reply packages/react/src/stories/ic-select.stories.mdx:510]: This is capitalized because it returns a React component.

**Comment by ad9242** [comment]
if using on a form with a reset button, then the searchable variant doesn't reset to the original value, only to null. Should it reset to the original value? Although I guess you would not necessarily have the original options passed in in order to set the selected item

**Review by ad9242 — DISMISSED** [review]

**Review by ad9242 — APPROVED** [review]

**Review by GCHQ-Developer-465 — APPROVED** [review]

**Pull request merged**